### PR TITLE
Support creating tensor from list of tensors

### LINKED ISSUE #232: How to perform all tensor on np.array packages
Example：
```
import mars.tensor as mt
import numpy as np

def test_x():
    X1 = mt.tensor([0, 1])
    X2 = mt.tensor([2, 3])
    X = np.array([X1,X2])
    print(X.execute())
```
How to calculate X？

### PR DESCRIPTION
**Pull request by sighingnow**
## What do these changes do?

Support creating tensor from list of tensors by creating a `stack` op on those tensors.

## Related issue number

Fixes #232

### CODE REVIEW

**Comment by chineking** [comment]
Thank you for your contribution, our habit is write the `fixes #xxx` in the **Related issue number**, and  not included in the title.

File: mars/tensor/expressions/datasource/array.py [hunk 1]
```diff
@@ -129,7 +129,8 @@ def tensor(data, dtype=None, chunk_size=None, gpu=None, sparse=False):
         if dtype is not None and data.dtype != dtype:
             return data.astype(dtype)
         return data
-    elif isinstance(data, tuple) and all(isinstance(d, TENSOR_TYPE) for d in data):
```
Comment (chineking) [inline mars/tensor/expressions/datasource/array.py]: Could be `isinstance(data, (tuple, list))`

**Comment by chineking** [comment]
The failure on PY35 of appveyor may happen sometimes due to unknown reason, rerun could make it work.

File: mars/tensor/expressions/datasource/array.py (hunk 1 above)
Comment (sighingnow) [reply mars/tensor/expressions/datasource/array.py]: Done.

**Comment by sighingnow** [comment]
Thanks for the suggestion! I have revised the title and the patch. I have also noticed the non-deterministic failure on appveyor and have no idea about the reason.

**Pull request merged**